pkgs/development/python-modules/yapf/default.nix

```
@@ -1,19 +1,56 @@
-{ lib, buildPythonPackage, fetchPypi }:
+{ lib
+, buildPythonApplication
+, fetchFromGitHub
+, nose
+, setuptools
+}:
 
-buildPythonPackage rec {
+buildPythonApplication rec {
   pname = "yapf";
-  version = "0.30.0";
+  version = "0.31.0";
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "3000abee4c28daebad55da6c85f3cd07b8062ce48e2e9943c8da1b9667d48427";
+  src = fetchFromGitHub {
+    owner = "google";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-Yfc1ES16MT6IpqzBD79F1U/d+32O7KKWYm7nfS8LqcQ=";
   };
 
+  nativeBuildInputs = [
+    setuptools
+  ];
+  checkInputs = [
+    nose
+  ];
+
+  checkPhase = ''
+    nosetests
+  '';
+
   meta = with lib; {
-    description = "A formatter for Python code.";
-    homepage    = "https://github.com/google/yapf";
-    license     = licenses.asl20;
-    maintainers = with maintainers; [ siddharthist ];
-  };
+    homepage = "https://github.com/google/yapf";
+    description = "Yet Another Python Formatter";
```

```suggestion
    description = "Yet Another Python Formatter";
    homepage    = "https://github.com/google/yapf";
```
We don't want to move around random stuff.